usage_registry.go

```diff
@@ -26,29 +31,47 @@ type UsageRegistry interface {
 	StartReporting(eventSink Handler, reportInterval time.Duration, msgQueueSize int)
 }
 
-func NewUsageRegistry(sourceId string, tags map[string]string, closeNotify <-chan struct{}) UsageRegistry {
+type UsageRegistryConfig struct {
+	SourceId             string
+	Tags                 map[string]string
+	EventQueueSize       int
+	CloseNotify          <-chan struct{}
+	IntervalAgeThreshold time.Duration
+}
+
+func DefaultUsageRegistryConfig(sourceId string, closeNotify <-chan struct{}) UsageRegistryConfig {
+	return UsageRegistryConfig{
+		SourceId:             sourceId,
+		Tags:                 map[string]string{},
+		CloseNotify:          closeNotify,
+		EventQueueSize:       DefaultEventQueueSize,
+		IntervalAgeThreshold: DefaultIntervalAgeThreshold,
+	}
+}
+
+func NewUsageRegistry(config UsageRegistryConfig) UsageRegistry {
+	if config.EventQueueSize < 16 {
```

Comment:
consts for the min values?

Reply:
I don't love constants when it makes it harder to follow. There are only two uses and they're right next to each other, so you can see what's happening. If you add a constant you've got to navigate around to understand the logic. 
Added one anyway :(